src/gnn_reco/components/utils.py

```diff
@@ -0,0 +1,266 @@
+
+import torch
+from tqdm import tqdm
+import pandas as pd
+import numpy as np
+from copy import deepcopy
+import torch
+from gnn_reco.data.sqlite_dataset import SQLiteDataset
+from sklearn.model_selection import train_test_split
+from torch_geometric.data.batch import Batch
+import os
+from sklearn.preprocessing import RobustScaler
+import sqlite3
+import pandas as pd
+import pickle
+import numpy as np
+import os
+
+class EarlyStopping(object):
```

Comment:
From here and to L155, most of this code could probably be imported from `pytorch-ligtning` (see e.g. https://pytorch-lightning.readthedocs.io/en/1.4.0/api/pytorch_lightning.callbacks.early_stopping.html; without me knowing the details of that package) which could reduce maintenance overhead. I think we should leave the code in this PR as-is, but am mainly commenting this to hear your thoughts on this, @RasmusOrsoe and @mhaminh.